### PR TITLE
Contract storage commands

### PR DESCRIPTION
**Pull request by ngdentdev**
Added contract storage list, contract storage get, and contract storage update commands to worknet.

### CODE REVIEW

File: src/worknet/Commands/ContractCommand.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+using McMaster.Extensions.CommandLineUtils;
+using Neo;
+using Neo.BlockchainToolkit.Models;
+
+namespace NeoWorkNet.Commands
+{
+  [Command("contract", Description = "Manage neo-worknet values")]
```
Comment (devhawk) [inline src/worknet/Commands/ContractCommand.cs:7]: Given that there aren't any other `contract` commands, I don't think the extra command verbosity is worth it. Lets make these just `storage get`, `storage list` and `storage update`

File: src/worknet/Node/WorkNetNode.cs [hunk 2]
```diff
@@ -7,255 +8,256 @@
...
 
 namespace NeoWorkNet.Node;
 
 class WorkNetNode
 {
-    readonly WorknetChain chain;
-    readonly string dataDirectory;
-
-    public WorkNetNode(WorknetChain chain, string filename)
+  readonly WorknetChain chain;
```
Comment (devhawk) [inline src/worknet/Node/WorkNetNode.cs:27]: Why did the spacing change from 4 spaces to 2? Can we standardize on 4 spaces please?

File: src/worknet/Commands/ContractCommand.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        var contractCount = contracts.Count(c => c.Name.Equals(contract, StringComparison.OrdinalIgnoreCase));
+        if (contractCount == 0) throw new Exception($"Cannot find contract: {contract}");
+        if (contractCount > 1) throw new Exception("Contract name is not unique. Please use contract hash instead.");
+        contractInfo = contracts.SingleOrDefault(c => c.Name.Equals(contract, StringComparison.OrdinalIgnoreCase));
+        contractHash = contractInfo?.Hash ?? UInt160.Zero;
+      }
+
+      if (contractHash == UInt160.Zero) throw new Exception("Invalid Contract argument");
+
+      contractInfo = contracts.SingleOrDefault(c => c.Hash == contractHash);
```
Comment (devhawk) [inline src/worknet/Commands/ContractCommand.cs:33]: Seems like this code should be in the TryParse else block. No need to search thru the list of contracts again if the user specified contract by name

File: src/worknet/Commands/ContractCommand.Storage.Get.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        internal string Key { get; init; } = string.Empty;
+        internal async Task<int> OnExecuteAsync(CommandLineApplication app, IConsole console)
+        {
+          try
+          {
+            var (chain, filename) = await fs.LoadWorknetAsync(app).ConfigureAwait(false);
+            var node = new WorkNetNode(chain, filename);
+            ContractInfo? contractInfo = ContractCommand.FindContractInfo(chain, Contract);
+            var writer = console.Out;
+            byte[]? value = node.GetStorage(contractInfo, Convert.FromHexString(Key.Substring(2)));
```
Comment (devhawk) [inline src/worknet/Commands/ContractCommand.Storage.Get.cs:40]: `0x` on key should be optional. Please check for it (case insensitive) before stripping it out via substring

File: src/worknet/Commands/ContractCommand.Storage.Get.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            var stringValue = value == null ? string.Empty : $"0x{Convert.ToHexString(value)}";
```
Comment (devhawk) [inline src/worknet/Commands/ContractCommand.Storage.Get.cs:41]: If there is no value for the specified key, command should say something like `key not found`

File: src/worknet/Commands/ContractCommand.Storage.Get.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+          this.fs = fileSystem;
+        }
+
+        [Argument(0, Description = "Contract name or hash")]
+        [Required]
+        internal string Contract { get; init; } = string.Empty;
+
+        [Argument(1, Description = "Key")]
+        [Required]
+        internal string Key { get; init; } = string.Empty;
```
Comment (devhawk) [inline src/worknet/Commands/ContractCommand.Storage.Get.cs:31]: can we add `--json` option to emit the info as JSON?

File: src/worknet/Commands/ContractCommand.Storage.List.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+      {
+        readonly IFileSystem fs;
+        public List(IFileSystem fileSystem)
+        {
+          this.fs = fileSystem;
+        }
+
+        [Argument(0, Description = "Contract name or hash")]
+        [Required]
+        internal string Contract { get; init; } = string.Empty;
```
Comment (devhawk) [inline src/worknet/Commands/ContractCommand.Storage.List.cs:27]: can we add a `--json` option to emit the results as JSON

File: src/worknet/Commands/ContractCommand.Storage.Update.cs [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+          {
+            var (chain, filename) = await fs.LoadWorknetAsync(app).ConfigureAwait(false);
+            byte[] keyBytes = GetKeyInBytes();
+            byte[] valueBytes = GetValueInBytes(chain);
+            var node = new WorkNetNode(chain, filename);
+            ContractInfo? contractInfo = ContractCommand.FindContractInfo(chain, Contract);
+            if(contractInfo.Id < 0){
+              throw new ArgumentException("Updating storage value for native Contracts are not allowed.");
+            }
+            node.UpdateValue(contractInfo, keyBytes, valueBytes);
```
Comment (devhawk) [inline src/worknet/Commands/ContractCommand.Storage.Update.cs:48]: Can update value fail?

File: src/worknet/Commands/ContractCommand.Storage.Update.cs [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+            return 0;
```
Comment (devhawk) [inline src/worknet/Commands/ContractCommand.Storage.Update.cs:49]: Should we have some text output on success or is the exit code good enough?

File: src/worknet/Commands/ContractCommand.Storage.Update.cs (hunk 8 above)
Comment (devhawk) [reply src/worknet/Commands/ContractCommand.Storage.Update.cs:48]: Never mind, put is void return

File: src/worknet/Commands/ContractCommand.Storage.List.cs (hunk 7 above)
Comment (ngdentdev) [reply src/worknet/Commands/ContractCommand.Storage.List.cs]: this is a sample output: {"script-hash":"0x50695ace7ab6c5f42d45828d36281bebd83201ee","storages":[{"key":"0xFF","value":"0xABCD"}]}

File: src/worknet/Commands/ContractCommand.Storage.Get.cs (hunk 6 above)
Comment (ngdentdev) [reply src/worknet/Commands/ContractCommand.Storage.Get.cs]: sample output: {"script-hash":"0x50695ace7ab6c5f42d45828d36281bebd83201ee","storages":[{"key":"0xFF","value":"0xABCD"}]} I made this same output format as the list command for consistency. The storage array will always be a length of 1. Let me know if that's ok.

File: src/worknet/Commands/ContractCommand.Storage.Update.cs (hunk 9 above)
Comment (ngdentdev) [reply src/worknet/Commands/ContractCommand.Storage.Update.cs]: yes add a text indicating the value is updated

File: src/worknet/Commands/ContractCommand.Storage.Get.cs (hunk 6 above)
Comment (devhawk) [reply src/worknet/Commands/ContractCommand.Storage.Get.cs]: I like the consistency

**Review by devhawk — APPROVED** [review]

**Pull request merged**